scripts/plugins/extauth-hook-AD.py

```diff
@@ -165,6 +165,9 @@ def _add_subject(self, subject_rec):
         try:
             sid = subject_rec['subject_identifier']
             name = subject_rec["other_config"]["subject-name"]
+            # CA-363207: Need to replace space to + for pam configuration
+            # "AA BB" -> "AA+BB", "AA   BB" -> "AA+++BB"
+            name = name.replace(' ', "+")
```

Comment:
Do we need to do the reverse and turn '+' into ' ' somewhere else?

Reply:
Apparently + is not a valid char for a logon name according to some Windows 2000 docs (can't easily find docs for anything newer, whether it has become valid meantime)

Reply:
windows server 2016 keep consistent with the doc.
put name inside [] works both for pbis and winbind, while replace ' ' -> '+' only work for pbis, 
updated the PR.

Reply:
@robhoes  I do not think we need to reverse